woocommerce/woocommerce-gutenberg-products-block
# Show feedback on buttons/checkout when processing

- Allows buttons to display a custom spinner when loading. I did not use `<spinner>` here because I wanted a different animation that would work better on buttons.
- Updates "Proceed to checkout", "Apply coupon", and "Place order" buttons to use spinners
- "Place order" button reflects checkout state and shows a checkmark when done before the redirect.
- Fields on checkout are disabled whilst processing is active.

Fixes #2172

### Screenshots

![2020-04-09 16 51 43](https://user-images.githubusercontent.com/90977/78914694-b0df9000-7a82-11ea-947c-fc9cecbef7dd.gif)
![2020-04-09 16 50 23](https://user-images.githubusercontent.com/90977/78914698-b210bd00-7a82-11ea-9147-c978222530f2.gif)

### How to test the changes in this Pull Request:

1. Go to checkout from the cart - button should show spinner once clicked
2. Apply a coupon. Button shows a spinner whilst loading
3. Place order. Watch button matches above gifs.
